code/rounds/InProgressRound.cs

```diff
@@ -104,20 +106,24 @@ public override void OnPlayerSpawn(TTTPlayer player)
             base.OnPlayerSpawn(player);
         }
 
-        private BaseRole IsRoundOver()
+        private TTTTeam IsRoundOver()
         {
-            bool innocentsAlive = Players.Exists((player) => player.Role is InnocentRole);
-            bool traitorsAlive = Players.Exists((player) => player.Role is TraitorRole);
+            List<TTTTeam> aliveTeams = new();
 
-            if (innocentsAlive && !traitorsAlive)
-            {
-                return new InnocentRole();
-            } else if (!innocentsAlive && traitorsAlive)
+            foreach (TTTPlayer player in Players)
             {
-                return new TraitorRole();
+                if (player.Team == null)
+                {
+                    continue;
+                }
+
+                if (!aliveTeams.Contains(player.Team))
+                {
+                    aliveTeams.Add(player.Team);
+                }
             }
 
-            return new NoneRole();
+            return aliveTeams.Count == 1 ? aliveTeams[0] : null;
```

Comment:
Are we able to have a `NoneTeam` similar to `NoneRole`? It would just allow us to have a `null`/`empty` state for Team so we can return that instead, just a bit cleaner imo.

Reply:
Thought about it, but that could interfer with the win condition. I'd maybe do this after the PR is finished, what do you think?

Reply:
Throw in a TODO, we can handle it later.

Reply:
Added it into the Trello.